src/components/Infra/InfraEnvWizard.tsx

```diff
@@ -1,164 +1,154 @@
 import { InfraWizard } from 'openshift-assisted-ui-lib';
-import { useK8sModel, k8sCreate, history, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk/api';
+import {
+  useK8sModel,
+  k8sCreate,
+  history,
+  useK8sWatchResource,
+} from '@openshift-console/dynamic-plugin-sdk/api';
 import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
 import { load } from 'js-yaml';
+import { RouteComponentProps } from 'react-router';
 
 import { InfraEnvKind } from '../../kind';
+import { getClusterDeployment, getPullSecretResource, parseStringLabels } from '../../k8s';
 
 import '../styles.scss';
 
-const InfraEnvWizard: React.FC = () => {
+const InfraEnvWizard: React.FC<RouteComponentProps<{ ns: string }>> = ({ match }) => {
   const [infraModel] = useK8sModel(InfraEnvKind);
   const [nmStateModel] = useK8sModel('agent-install.openshift.io~v1beta1~NMStateConfig');
   const [secretModel] = useK8sModel('core~v1~Secret');
   const [clusterDepModel] = useK8sModel('hive.openshift.io~v1~ClusterDeployment');
-  const [agentClusterInstallModel] = useK8sModel('extensions.hive.openshift.io~v1beta1~AgentClusterInstall');
+  const [agentClusterInstallModel] = useK8sModel(
+    'extensions.hive.openshift.io~v1beta1~AgentClusterInstall',
+  );
+  const namespace = match?.params?.ns || 'assisted-installer';
   const [infraEnvs] = useK8sWatchResource<K8sResourceCommon[]>({
     kind: InfraEnvKind,
-    namespace: 'assisted-installer',
+    namespace,
     isList: true,
   });
   const usedNames = infraEnvs.map((env) => env.metadata.name);
-  const onSubmit = React.useCallback(async (values) => {
-    await k8sCreate(secretModel, {
-      kind: 'Secret',
-      apiVersion: 'v1',
-      metadata: {
-        name: values.name,
-        namespace: 'assisted-installer'
-      },
-      data: {
-        '.dockerconfigjson': btoa(values.pullSecret),
-      },
-      type: 'kubernetes.io/dockerconfigjson',
-    });
+  const onSubmit = React.useCallback(
+    async (values) => {
+      const labels = parseStringLabels(values.labels);
 
-    const labels = values.labels.reduce((acc,curr) => {
-      const label = curr.split('=');
-      acc[label[0]] = label[1];
-      return acc;
-    }, {});
-    await k8sCreate(clusterDepModel, {
-      apiVersion: 'hive.openshift.io/v1',
-      kind: 'ClusterDeployment',
-      metadata: {
-        name: values.name,
-        namespace: 'assisted-installer',
-      },
-      spec: {
-        baseDomain: values.baseDomain,
-        clusterInstallRef:{
-          group: 'extensions.hive.openshift.io',
-          kind: 'AgentClusterInstall',
+      const secret = await k8sCreate(
+        secretModel,
+        getPullSecretResource({ namespace, name: values.name, pullSecret: values.pullSecret }),
+      );
+      await k8sCreate(
+        clusterDepModel,
+        getClusterDeployment({
+          namespace,
+          pullSecretName: secret?.metadata?.name,
+          ...values,
+        }),
+      );
```

Comment:
this needs to be 
```suggestion
        getClusterDeployment({
          namespace,
          pullSecretName: secret?.metadata?.name,
          ...values,
          labels,
        }),
      );
```
to pass the object labels from https://github.com/rawagner/dynamic-cim/pull/9/files#diff-9a7efbcdf1e2bcecc787c5040b867e71ba4851c74d2138b427d19818714e0667R35 , not the array labels from form

Reply:
done